src/Game/AccountManagement.java

```diff
@@ -41,45 +44,28 @@ public static void register(String user, String pass) {
 				int port = Replay.connection_port;
 				StreamUtil.initializeStream(Client.server_address, port);
 				StreamUtil.setStreamMaxRetries(Client.maxRetries);
-				byte limit30 = 0;
-				String param = Client.getParameter("limit30");
-				if (param != null && param.equals("1")) {
-					limit30 = 1;
-				}
 				
-				int[] keys = new int[]{
-						(int)(9.9999999E7 * Math.random()),
-						(int)(9.9999999E7 * Math.random()),
-						(int)(9.9999999E7 * Math.random()),
-						(int)(9.9999999E7 * Math.random())
-				};
 				StreamUtil.newPacket(2);
-				Object buffer = StreamUtil.getStreamBuffer();
-				StreamUtil.putIntTo(buffer, Client.version);
-				Object block = StreamUtil.getNewBuffer(500);
-				StreamUtil.putByteTo(block, (byte)11); //register packet encrypted id
-				StreamUtil.putIntTo(block, keys[0]); // isaac/xtea keys
-				StreamUtil.putIntTo(block, keys[1]);
-				StreamUtil.putIntTo(block, keys[2]);
-				StreamUtil.putIntTo(block, keys[3]);
-				StreamUtil.putStrTo(block, formatPass); // password
+                Object buffer = StreamUtil.getStreamBuffer();
+                StreamUtil.putShortTo(buffer, (short)(Client.version & 0xFFFF));
+
+                // Put Username
+                long formatUser = Util.username2hash(Client.username_login);
+                StreamUtil.putLongTo(buffer, formatUser);
+
+                // Put Password
+                enc_cred_put(buffer, formatPass, 0);

```

Comment:
The 0 here is sessionID. Make it a variable.

Reply:
done